server/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -444,17 +441,17 @@ public ActionModule(Settings settings, IndexNameExpressionResolver indexNameExpr
             UnaryOperator<RestHandler> newRestWrapper = plugin.getRestHandlerWrapper(threadPool.getThreadContext());
             if (newRestWrapper != null) {
                 logger.debug("Using REST wrapper from plugin " + plugin.getClass().getName());
+                if (plugin.getClass().getCanonicalName() == null ||
+                    plugin.getClass().getCanonicalName().startsWith("org.elasticsearch.xpack") == false) {
+                    logger.error("The " + plugin.getClass().getName() + " plugin tried to install a custom REST wrapper. This " +
+                        "functionality is not available anymore.");
+                    throw new IllegalArgumentException("The " + plugin.getClass().getName() + " plugin tried to install a custom REST " +
+                        "wrapper. This functionality is not available anymore.");
+                }
```

Comment:
Per https://github.com/elastic/elasticsearch/pull/73329, there should be tests for this functionality in ActionModuleTests.
Core shouldn't rely on x-pack to do its testing for it.

Reply:
Moved it to ActionModuleTests